### PR TITLE
[proxima-direct-io-kafka] #195 implement topicPattern for kafka URI

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java [hunk 1]
```diff
@@ -356,16 +381,24 @@ private void submitConsumerWithObserver(
...
-            if (log.isDebugEnabled()) {
-              log.debug("End offsets of current assignment {}: {}", kafka.assignment(), endOffsets);
-            }
+              if (log.isDebugEnabled()) {
+                log.debug(
+                    "End offsets of current assignment {}: {}", kafka.assignment(), endOffsets);
+              }
 
-            listener.onPartitionsAssigned(kafka.assignment());
+              listener.onPartitionsAssigned(kafka.assignment());
```
Comment (tisonet) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:396]: Why is needed this cycle?

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Utils.java [hunk 2]
```diff
@@ -19,36 +19,55 @@
...
 
+  static final String TOPIC_PATTERN_QUERY = "topicPattern";
+
   /**
    * Retrieve topic from given URI.
    *
    * @param uri the URL
    * @return topic name
    */
+  @Nullable
```
Comment (tisonet) [inline direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Utils.java:37]: Maybe Optional<String> would be better return type for this method.

**Review by tisonet — APPROVED** [review]

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java (hunk 1 above)
Comment (je-ik) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java:396]: Assignment of partitions for subscriptions with regex apparently takes multiple calls to poll().

**Review by dmvk — APPROVED** [review]
Nice, looking forward to trying this out 👍

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Utils.java (hunk 2 above)
Comment (je-ik) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Utils.java:37]: That is internal API, I like better to use Optionals on public methods only.

**Pull request merged**